test/functional/feature_reindex.py

```diff
@@ -68,12 +72,49 @@ def find_block(b, start):
             'LoadExternalBlockFile: Out of order block',
             'LoadExternalBlockFile: Processing out of order child',
         ]):
-            extra_args = [["-reindex"]]
-            self.start_nodes(extra_args)
+            self.start_node(0, extra_args=["-reindex"])
 
         # All blocks should be accepted and processed.
         assert_equal(self.nodes[0].getblockcount(), 12)
 
+    def reindex_readonly(self):
+        self.connect_nodes(0, 1)
+        addr = self.nodes[1].get_deterministic_priv_key().address
+
+        # generate enough blocks to ensure that the -fastprune node fills up the
+        # first blk00000.dat file and starts another block file
+
+        # How big are empty regtest blocks?
+        block_hash = self.generatetoaddress(self.nodes[1], 1, addr)[0]
+        block_size = self.nodes[1].getblock(block_hash)["size"]
+        block_size += 8 # BLOCK_SERIALIZATION_HEADER_SIZE
+
+        # How many blocks do we need to roll over a new .blk file?
+        block_count = self.nodes[1].getblockcount()
+        fastprune_blockfile_size = 0x10000
+        size_needed = fastprune_blockfile_size - (block_size * block_count)
+        blocks_needed = size_needed // block_size
```

Comment:
I like the way you're calculating the number of blocks needed rather than hard-coding the number; I verified that this value (when I run the test) is 240, which matches what @jonatack suggested.